docs/metrics/selfcheckgpt.md

```diff
@@ -0,0 +1,157 @@
+
+#### Code Example
+
+The scores we compute are at sentence- and document-level where values lie between `[0.0, +inf)` with a higher value
+suggesting that the response may be non-factual.
+
+```py
+selfcheck_ngram = SelfCheckNgram(n=1) # n=1 means Unigram, n=2 means Bigram, etc.
+
+sent_scores_ngram = selfcheck_ngram.predict(
+    sentences = [main_answer],
+    passage = main_answer,
+    sampled_passages = [answer],
+)
+print(sent_scores_ngram)
+```
+
+```
+{'sent_level': { # sentence-level score similar to MQAG and BERTScore variant
+    'avg_neg_logprob': [2.9464540757764373],
+    'max_neg_logprob': [3.828641396489095]
+    },
+'doc_level': { # document-level score such that avg_neg_logprob is computed over all tokens
+    'avg_neg_logprob': 2.9464540757764373,
+    'avg_max_neg_logprob': 3.828641396489095
+    }
+}
+```
+
+
+### LLM Prompt-based Consistency Score
+
+With this approach, SelfCheckGPT utilizes an LLM such as `gpt-3.5-turbo` and a prompt to compute a boolean consistency
+score where `Yes` implies that the context and sentence are consistent (no hallucination) and otherwise, `No`, for each sample
+answer which is passed as the context in our prompts.
+
+#### Code Example
+
+We define a `PROMPT`
+
+```
+Context: {answer}
+Sentence: {main_answer}
+Is the sentence supported by the context above?
+Answer Yes or No. If the answer is No, on the next line, explain in about ten words why it is not supported by the context.
+```
+
+Then, using OpenAI's API we make a query
+
+```py
+response = openai.chat.completions.create(
+    model="gpt-3.5-turbo",
+    messages=[
+        {
+        "role": "user",
+        "content": PROMPT
+        },
+    ],
+    temperature=0,
+    max_tokens=50,
+    top_p=1,
+    frequency_penalty=0,
+    presence_penalty=0
+)
+
+if len(response.choices) < 1:
+    return (None, "")
+
+response = str(response.choices[0].message.content)
+
+return (response == "Yes" or response == "Yes.", response.split("\n")[-1] if response != "Yes" and response != "Yes." else "")
+```
+
+we can expect output like the below
+
+```
+No
+The sentence is not supported by the context because it states that the spiciest part of a chili pepper is the white pith,
+also known as the placenta, which is not mentioned in the context.
+```
+
+To obtain a numerical consistency score we perform the same operation but on different context answers and aggregate across the
+returned boolean consistency scores.
+
+## Limitations and Advantages
+
+SelfCheckGPT is a very powerful black-box evaluation approach thats advantages come from its overall lower resource usage as
+well as its performance despite only having access to text-based responses. However, SelfCheckGPT can often be mislead by sentences
+that contain consistent by factually incorrect statements and for the best variant of SelfCheckGPT with LLM Prompt, it can also be
+computationally heavy.
```

Comment:
For LLM Prompt, the same thing as the llm-prompt-based-metric can be said about its limitations. can be costly and data privacy can be an issue.